DFReader.py

```diff
@@ -256,16 +279,10 @@ def rewind_event(self):
 
     def message_arrived(self, m):
         type = m.get_type()
-        if not type in self.counts:
-            self.counts[type] = 1
-        else:
-            self.counts[type] += 1
+        self.counts[type] = self.counts(type, 0) + 1
```

Comment:
This doesn't actually work :-)

Were you thinking of using an accessor into the dict here?